packages/data-context/src/actions/ElectronActions.ts

```diff
@@ -63,4 +68,46 @@ export class ElectronActions {
   showItemInFolder (url: string) {
     this.ctx.electronApi.showItemInFolder(url)
   }
+
+  showOpenDialog () {
+    const props: OpenDialogOptions = {
+      // we only want the user to select a single
+      // directory. not multiple, and not files
+      properties: ['openDirectory'],
+    }
+
+    return this.ctx.electronApi.showOpenDialog(props)
+    .then((obj) => {
+      // return the first path since there can only ever
+      // be a single directory selection
+      return _.get(obj, ['filePaths', 0])
+    })
+  }
+
+  showSaveDialog (integrationFolder: string) {
+    // Do we want to attach browserWindow (?)
+    assert(this.electron.browserWindow, 'Browser window is not set')
```

Comment:
Added the browser window in the meantime,  if needed we can update it to receive it from the args or the launchpad